apps/api/src/app/events/usecases/map-trigger-recipients/map-trigger-recipients.use-case.ts

```diff
@@ -0,0 +1,139 @@
+import { Injectable } from '@nestjs/common';
+import {
+  ISubscribersDefine,
+  ITopic,
+  TriggerRecipient,
+  TriggerRecipientSubscriber,
+  TriggerRecipientTopics,
+  TriggerRecipients,
+  TriggerRecipientsPayload,
+} from '@novu/node';
+import {
+  EnvironmentId,
+  LogCodeEnum,
+  LogStatusEnum,
+  OrganizationId,
+  TopicId,
+  TriggerRecipientsTypeEnum,
+  UserId,
+} from '@novu/shared';
+
+import { MapTriggerRecipientsCommand } from './map-trigger-recipients.command';
+
+import { CreateLog, CreateLogCommand } from '../../../logs/usecases/create-log';
+import { GetTopicSubscribersCommand, GetTopicSubscribersUseCase } from '../../../topics/use-cases';
+
+interface ILogTopicSubscribersPayload {
+  environmentId: EnvironmentId;
+  organizationId: OrganizationId;
+  topicId: TopicId;
+  transactionId: string;
+  userId: UserId;
+}
+
+@Injectable()
+export class MapTriggerRecipients {
+  constructor(private createLog: CreateLog, private getTopicSubscribers: GetTopicSubscribersUseCase) {}
+
+  async execute(command: MapTriggerRecipientsCommand): Promise<ISubscribersDefine[]> {
+    const { environmentId, organizationId, recipients, transactionId, userId } = command;
+
+    const mappedRecipients = Array.isArray(recipients) ? recipients : [recipients];
+
+    const simpleSubscribers: ISubscribersDefine[] = this.findSubscribers(mappedRecipients);
+
+    const topicSubscribers: ISubscribersDefine[] = await this.getSubscribersFromAllTopics(
+      transactionId,
+      environmentId,
+      organizationId,
+      userId,
+      mappedRecipients
+    );
+
+    return [...simpleSubscribers, ...topicSubscribers];
```

Comment:
I'm wondering if should we care about the scenario when there will be subscriber duplicates from the topic + request?
Imagine the request body like this:
```
{
   to: ['subscriberId', { key: 'some_topic', type: TriggerRecipientsTypeEnum.SUBSCRIBER }]
}
```
then the `simpleSubscribers and topicSubscribers` arrays will look like:
```
const simpleSubscribers = [{ subscriberId: 'subscriberId' }]
const topicSubscribers = [{ subscriberId: 'subscriberId' }]
```

And in a result, there will be two same subscribers.

WDYT?

Reply:
Yes! That edge case and others are pending to be done in forthcoming PRs. First I had to see this working. 😅
But it was a good reminder to create the task.